extensions/amp-access-laterpay/0.1/amp-access-laterpay.js

```diff
@@ -28,28 +87,366 @@ export class LaterpayVendor {
+    ) ? this.laterpayConfig_.configUrl
+      : CONFIG_URL;
+
+    /** @private {string} */
+    this.currentLocale_ = this.laterpayConfig_.locale || 'en';
+
+    /** @private {Object} */
+    this.i18n_ = Object.assign(DEFAULT_MESSAGES,
+                  this.laterpayConfig_.localeMessages || {});
+
+    /** @private {string} */
+    this.purchaseConfigBaseUrl_ = configUrl + CONFIG_BASE_PATH;
+    const articleId = this.laterpayConfig_.articleId;
+    if (articleId) {
+      this.purchaseConfigBaseUrl_ += '&article_id=' + articleId;
+    }
+
+    /** @const @private {!Timer} */
+    this.timer_ = timerFor(this.win_);
+
+    /** @const @private {!Vsync} */
+    this.vsync_ = vsyncFor(this.win_);
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.win_);
+
+    installStyles(this.win_.document, CSS, () => {}, false, TAG);
   }
 
   /**
    * @return {!Promise<!JSONType>}
    */
   authorize() {
-    const win = this.accessService_.win;
-    user().assert(isExperimentOn(win, 'amp-access-laterpay'),
+    user().assert(isExperimentOn(this.win_, TAG),
         'Enable "amp-access-laterpay" experiment');
-    // TODO: implement
-    return Promise.resolve({access: true});
+    return this.getPurchaseConfig_()
+    .then(response => {
+      if (response.status === 204) {
+        throw user()
+          .createError('No merchant domains have been matched for this ' +
+            'article, or no paid content configurations are setup.');
+      }
+
+      if (this.laterpayConfig_.scrollTopAfterAuth) {
+        this.vsync_.mutate(() => this.viewport_.setScrollTop(0));
+      }
+      this.emptyContainer_();
+      return {access: response.access};
+    }, err => {
+      const status = err && err.response && err.response.status;
+      if (status === 402) {
+        this.purchaseConfig_ = err.responseJson;
+        // empty before rendering, in case authorization is being called again
+        // with the same state
+        this.emptyContainer_()
+          .then(this.renderPurchaseOverlay_.bind(this));
+      } else {
+        throw err;
+      }
+      return {access: err.responseJson.access};
+    });
+  }
+
+  /**
+   * @return {!Promise<Object>}
+   * @private
+   */
+  getPurchaseConfig_() {
+    const url = this.purchaseConfigBaseUrl_ +
+                '&article_title=' + encodeURIComponent(this.getArticleTitle_());
+    const urlPromise = this.accessService_.buildUrl(
+      url, /* useAuthData */ false);
+    return urlPromise.then(url => {
+      dev().fine(TAG, 'Authorization URL: ', url);
+      return this.timer_.timeoutPromise(
+          AUTHORIZATION_TIMEOUT,
+          this.xhr_.fetchJson(url, {
+            credentials: 'include',
+            requireAmpResponseSourceOrigin: true,
+          }));
+    });
+  }
+
+  /**
+   * @return {!string}
+   * @private
+   */
+  getArticleTitle_() {
+    const title = this.doc_.querySelector(
+      this.laterpayConfig_.articleTitleSelector);
+    dev().assert(
```

Comment:
Right, I actually was unsure of that and forgot to ask.

I believe it is strictly required because we show it in the payment dialog (so the user knows what they're paying for) but I'll look into what we do in our existing frontend integration.